### PR TITLE
GEODE-8687: Fix for handling of PdxSerializationException on client

### PR DESCRIPTION
**Pull request by jvarenina**
* Whenever PdxSerializationException is thrown on client for problematic CQ event, client will try to log that exception
  together with the problematic event value. But the problem is that client has to de-serialize event value in order to log it.
  When client tries to do that, it gets another PdxSerializationException which is now catched at higher level and cause
  shutdown of CacheClientUpdater and subscription queue connection.

* Behavior with this fix: At the reception of event that provoke
  PdxSerializationException client will only log the exception and ignore
  PdxSerializationException that happen during logging

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by Bill** [comment]
I don't know this area of the code yet. I'm going to see if I can find some other reviewers @kohlmu-pivotal

File: geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientCQAutoSerializer.java [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+import org.apache.geode.test.dunit.Invoke;
+import org.apache.geode.test.dunit.rules.ClientVM;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.junit.categories.ClientSubscriptionTest;
+import org.apache.geode.test.junit.categories.SerializationTest;
+import org.apache.geode.test.junit.rules.GfshCommandRule;
+
+@Category({ClientSubscriptionTest.class, SerializationTest.class})
+public class DurableClientCQAutoSerializer implements Serializable {
```
Comment (mkevo) [inline geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientCQAutoSerializer.java:53]: Please rename this file to be clear that it is the test. Something like _DurableClientCQAutoSerializerDUnitTest_

File: geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientCQAutoSerializer.java [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    locator.invoke(() -> {
+      ClusterStartupRule.memberStarter
+          .waitUntilRegionIsReadyOnExactlyThisManyServers(SEPARATOR + REPLICATE_REGION_NAME, 2);
+      ClusterStartupRule.memberStarter
+          .waitUntilRegionIsReadyOnExactlyThisManyServers(SEPARATOR + PARTITION_REGION_NAME, 2);
+    });
+  }
+
+  @Test
+  public void correctClassPathsAutoSerializer()
```
Comment (mkevo) [inline geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientCQAutoSerializer.java:122]: Please change the test name. Something like _testcorrectClassPathsAutoSerializer_

File: geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientCQAutoSerializer.java [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    startDataProvisionClient(TEST_OBJECT1_CLASS_PATH, TEST_OBJECT2_CLASS_PATH);
+    provisionRegionsWithData();
+
+    // Check that all events are received and successfully deserialized in cq listener
+    checkCqEvents(LIST_TEST_OBJECT1.size(), LIST_TEST_OBJECT2.size());
+    verifyThatOnlyOneServerHostDurableSubscription();
+  }
+
+  @Test
+  public void faultyClassPathAutoSerializer()
```
Comment (mkevo) [inline geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientCQAutoSerializer.java:142]: The same as above. Something like _testfaultyClassPathAutoSerializer_

File: geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/TestAutoSerializerObject1.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+public class TestAutoSerializerObject1 implements PdxSerializerObject {
+  protected String data1;
+  protected String data2;
+  protected int numData3;
+
+  public TestAutoSerializerObject1() {
+    this("", "", 0);
+  }
+
+  protected TestAutoSerializerObject1(String data1, String data2, int numData3) {
```
Comment (mkevo) [inline geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/TestAutoSerializerObject1.java:34]: Please change this _numData3_ to _numData_, also change get methods.

File: geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/TestAutoSerializerObject1.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  @Override
+  public boolean equals(Object o) {
+
+    if (this == o)
+      return true;
+
+    if (o == null)
+      return false;
+
+    if (getClass() != o.getClass())
```
Comment (mkevo) [inline geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/TestAutoSerializerObject1.java:96]: Can you merge this if statement with one above as they returning the same value?

File: geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/TestAutoSerializerObject2.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+public class TestAutoSerializerObject2 implements PdxSerializerObject {
+  protected String data1;
+  protected String data2;
+  protected int numData3;
+
+  public TestAutoSerializerObject2() {
+    this("", "", 0);
+  }
+
+  protected TestAutoSerializerObject2(String data1, String data2, int numData3) {
```
Comment (mkevo) [inline geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/TestAutoSerializerObject2.java:34]: Please change this _numData3_ to _numData_, also change get methods.

File: geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/TestAutoSerializerObject2.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o)
+      return true;
+
+    if (o == null)
+      return false;
+
+    if (getClass() != o.getClass())
```
Comment (mkevo) [inline geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/TestAutoSerializerObject2.java:95]: Can you merge this if statement with one above as they returning the same value?

**Review by mkevo — CHANGES_REQUESTED** [review]

File: geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientCQAutoSerializer.java (hunk 1 above)
Comment (jvarenina) [reply geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientCQAutoSerializer.java]: Hi Mario, thanks for the comments. I have applied all of them in the following commit.

Comment (mkevo) [reply geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientCQAutoSerializer.java]: Thanks Jakov!

**Review by mkevo — APPROVED** [review]

**Review by mivanac — APPROVED** [review]

**Pull request merged**